components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java

```diff
@@ -302,53 +321,37 @@ public DropboxMoveResult move(String remotePath, String newRemotePath) throws Dr
 
     /**
      * Get the content of every file inside the remote path.
+     *
      * @param remotePath the remote path where to download from
-     * @return a result object with the content (ByteArrayOutputStream) of every files inside the remote path.
+     * @return a result object with the content (ByteArrayOutputStream) of every
+     *         files inside the remote path.
      * @throws DropboxException
      */
     public DropboxFileDownloadResult get(String remotePath) throws DropboxException {
         return new DropboxFileDownloadResult(downloadFilesInFolder(remotePath));
     }
 
-    /**
-     * @deprecated not in use
-     */
-    @Deprecated
-    public boolean isDirectory(String path) throws DropboxException {
-        try {
-            DbxEntry.WithChildren listing = client.getMetadataWithChildren(path);
-            return listing.children != null;
-        } catch (DbxException e) {
-            throw new DropboxException(path + " does not exist or can't obtain metadata");
-        }
-    }
-
-
     private Map<String, Object> downloadFilesInFolder(String path) throws DropboxException {
         try {
-            DbxEntry.WithChildren listing = client.getMetadataWithChildren(path);
-            if (listing == null) {
-                return Collections.emptyMap();
-            } else if (listing.children == null) {
-                LOG.debug("downloading a single file...");
-                Map.Entry<String, Object> entry = downloadSingleFile(path);
-                return Collections.singletonMap(entry.getKey(), entry.getValue());
+            ListFolderResult folderResult = client.files().listFolder(path.equals("/") ? "" : path);
+            Map<String, Object> returnMap = new LinkedHashMap<>();
+            for (Metadata entry : folderResult.getEntries()) {
+                returnMap.put(entry.getPathDisplay(), downloadSingleFile(entry.getPathDisplay()).getValue());
             }
-            Map<String, Object> result = new HashMap<>();
-            for (DbxEntry entry : listing.children) {
-                if (entry.isFile()) {
-                    try {
-                        Map.Entry<String, Object> singleFile = downloadSingleFile(entry.path);
-                        result.put(singleFile.getKey(), singleFile.getValue());
-                    } catch (DropboxException e) {
-                        LOG.warn("Cannot download from path={}, reason={}. This exception is ignored.", entry.path, e.getMessage());
-                    }
+            return returnMap;
+        } catch (ListFolderErrorException e) {
```

Comment:
Can you explain why you catch this exception and then try again?

Reply:
This was based on my suggestion. I tried different ways but we don't have the method to get metadata that comes with a flag to fulfill isFile/isDirectory. It assumes you know what you are trying to get. 
So the idea is to run listFolder, and if you get that Exception, the folder doesn't exist or it's a file.
It could be replaced by a new parameter and avoid the try/catch, but that would break current folder endpoints.

I agree that this should be released in the next iteration. V1 was retired and the released component is completely broken.

If you guys need something, let me know.

Reply:
Thank you very much for the explanation. So lets go with this. And we can add a code comment that this code is on purpose.